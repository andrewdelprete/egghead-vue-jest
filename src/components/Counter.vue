<template>
  <div>
    <div><span class="count" data-testid="count">{{ multiplierCount }}</span></div>
    <button @click="increment">Increments</button>
    <button @click="decrement">Decrements</button>
    <button @click="reset">Reset</button>
  </div>
</template>

<script>
export default {
  props: {
    multiplier: { type: Number, default: 1 }
  },
  data() {
    return {
      count: 0
    };
  },
  methods: {
    increment() {
      this.count++;
    },
    decrement() {
      this.count--;
    },
    reset() {
      this.count = 0;
    }
  },
  computed: {
    multiplierCount: function() {
      return this.count * this.multiplier;
    }
  }
};
</script>
