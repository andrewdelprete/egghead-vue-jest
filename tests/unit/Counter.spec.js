import { mount } from "@vue/test-utils";
import Counter from "@/components/Counter.vue";

describe("Counter.vue", () => {
  it("renders component", () => {
    const wrapper = mount(Counter);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("has initial count of 0", () => {
    const wrapper = mount(Counter);
    expect(wrapper.vm.count).toEqual(0);
  });

  it("increments count by 1", () => {
    const wrapper = mount(Counter);
    wrapper.vm.increment();
    expect(wrapper.vm.count).toBe(1);
  });

  it("decrements count by 1", () => {
    const wrapper = mount(Counter);
    wrapper.vm.decrement();
    expect(wrapper.vm.count).toBe(-1);
  });

  it("resets count to 0", () => {
    const wrapper = mount(Counter);
    wrapper.vm.reset();
    expect(wrapper.vm.count).toBe(0);
  });

  it("computed prop is incremented", () => {
    const wrapper = mount(Counter, {
      propsData: { multiplier: 2 }
    });

    wrapper.vm.increment();
    expect(wrapper.vm.multiplierCount).toBe(2);
    wrapper.vm.increment();
    expect(wrapper.vm.multiplierCount).toBe(4);
  });

  it("computed prop is incremented", () => {
    const wrapper = mount(Counter, {
      propsData: { multiplier: 4 }
    });

    wrapper.vm.decrement();
    expect(wrapper.vm.multiplierCount).toBe(-4);
    wrapper.vm.decrement();
    expect(wrapper.vm.multiplierCount).toBe(-8);
  });
});
