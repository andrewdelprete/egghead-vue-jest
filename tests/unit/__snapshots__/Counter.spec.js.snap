// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue renders component 1`] = `
<div>
  <div>
    <span
      class="count"
    >
      0
    </span>
  </div>
   
  <button>
    Increments
  </button>
   
  <button>
    Decrements
  </button>
   
  <button>
    Reset
  </button>
</div>
`;
